pkg/controller/logstorage/secrets/secret_controller.go

```diff
@@ -0,0 +1,619 @@
+	// redundant because all certificates will be signed by the root CA. However, in some scenarios custom certificates may
+	// be used, in which case they must be present in the bundle.
+	//
+	// Each entry is the name of the secret, and the value is the Namespace to query it from.
+	certs := map[string]string{
+		// Get certificate for TLS on Prometheus metrics endpoints. This is created in the monitor controller.
+		monitor.PrometheusClientTLSSecretName: common.OperatorNamespace(),
+
+		// Get certificate for es-proxy, which Linseed and es-gateway need to trust.
+		render.ManagerTLSSecretName: req.TruthNamespace(),
+
+		// Get certificate for fluentd, which Linseed needs to trust in a standalone or management cluster.
+		render.FluentdPrometheusTLSSecretName: common.OperatorNamespace(),
+
+		// Get certificate for intrusion detection controller, which Linseed needs to trust in a standalone or management cluster.
+		render.IntrusionDetectionTLSSecretName: common.OperatorNamespace(),
+		render.AnomalyDetectorTLSSecretName:    common.OperatorNamespace(),
+
+		// Get certificate for DPI, which Linseed needs to trust in a standalone or management cluster.
+		render.DPITLSSecretName: common.OperatorNamespace(),
+
+		// Get compliance certificates, which Linseed needs to trust.
+		render.ComplianceServerCertSecret:  common.OperatorNamespace(),
+		render.ComplianceSnapshotterSecret: common.OperatorNamespace(),
+		render.ComplianceBenchmarkerSecret: common.OperatorNamespace(),
+		render.ComplianceReporterSecret:    common.OperatorNamespace(),
+
+		// Get certificate for policy-recommendation, which Linseed needs to trust.
+		render.PolicyRecommendationTLSSecretName: common.OperatorNamespace(),
+
+		// Linseed and es-gateway need to trust the
+		render.TigeraElasticsearchInternalCertSecret: common.OperatorNamespace(),
+
+		// es-gateay and Linseed need to trust certificates signed by the root tigera-operator CA.
+		// TODO: Why is this needed in addition to the elasticsearch internal cert above?
+		certificatemanagement.CASecretName: common.OperatorNamespace(),
+	}
+	for certName, certNamespace := range certs {
+		cert, err := cm.GetCertificate(r.client, certName, certNamespace)
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceReadError, "Failed to get certificate", err, log)
+			return nil, err
+		} else if cert == nil {
+			msg := fmt.Sprintf("%s/%s secret not available yet, will add it if/when it becomes available", certNamespace, certName)
+			log.Info(msg)
+		} else {
+			log.V(2).Info("Adding upstream certificate", "cert", certName)
+			collection.upstreamCerts = append(collection.upstreamCerts, cert)
+		}
+	}
+
+	// Add the certificate manager's keypair. This ensures that the secret used to sign certificates and keys
+	// in this tenant's namespace is saved for other controllers to use.
+	// TODO: This might not be the right place for this.
```

Comment:
Right now, for single-tenant clusters, the root CA is installed into the tigera-operator namespace by the core controller. However, the core controller is single-tenant, and so we need to do this somewhere else for multi-tenant systems, and ideally wherever we do it also works for single-tenant systems so we're not duplicating logic.

I am thinking this might just be a new "Root CA controller" (it may be obvious that I am on a 'write new controllers' kick right now).